src/test/scala/chiselTests/BoringUtilsSpec.scala

```diff
@@ -125,4 +126,145 @@ class BoringUtilsSpec extends ChiselFlatSpec with ChiselRunners {
     runTester(new InternalBoreTester, annotations = TesterDriver.verilatorOnly) should be(true)
   }
 
+  it should "work using new API" in {
+    class Baz extends RawModule {
+      val a_wire = WireInit(UInt(1.W), DontCare)
+      dontTouch(a_wire)
+    }
+    class Bar extends RawModule {
+      val b_wire = WireInit(UInt(2.W), DontCare)
+      dontTouch(b_wire)
+
+      val baz = Module(new Baz)
+    }
+    class Foo extends RawModule {
+      val a = IO(Output(UInt()))
+      val b = IO(Output(UInt()))
+      val c = IO(Output(UInt()))
+
+      val c_wire = WireInit(UInt(3.W), DontCare)
+      dontTouch(c_wire)
+
+      val bar = Module(new Bar)
+
+      a := BoringUtils.bore(bar.baz.a_wire)
+      b := BoringUtils.bore(bar.b_wire)
+      c := BoringUtils.bore(c_wire)
+    }
+    //println(circt.stage.ChiselStage.emitCHIRRTL(new Foo, args))
+    (circt.stage.ChiselStage
+      .emitCHIRRTL(new Foo)
+      .split('\n')
+      .map(_.takeWhile(_ != '@'))
+      .map(_.trim) should contain).inOrder(
+      "module Baz :",
+      "output a_bore : UInt<1>",
+      "a_bore <= a_wire",
+      "module Bar :",
+      "output b_bore : UInt<2>",
+      "a_bore <= baz.a_bore",
+      "b_bore <= b_wire",
+      "module Foo :",
+      "a <= a_bore",
+      "b <= b_bore",
+      "c <= c_wire",
+      "a_bore <= bar.a_bore",
+      "b_bore <= bar.b_bore"
+    )
+  }
+
+  it should "not work over a Definition/Instance boundary" in {
+    import chisel3.experimental.hierarchy._
+    @instantiable
+    class Bar extends RawModule {
+      @public val a_wire = WireInit(UInt(1.W), DontCare)
+      dontTouch(a_wire)
+    }
+    class Foo extends RawModule {
+      val a = IO(Output(UInt()))
+
+      val bar = Instance(Definition((new Bar)))
+
+      a := BoringUtils.bore(bar.a_wire)
+    }
+    val e = intercept[Exception] {
+      circt.stage.ChiselStage.emitCHIRRTL(new Foo, args)
+    }
+    e.getMessage should include("Cannot bore across a Definition/Instance boundary")
+  }
+
+  it should "work boring upwards" in {
+    import chisel3.experimental.hierarchy._
+    class Bar(parentData: Data) extends RawModule {
+      val q = Wire(UInt(1.W))
+      q := BoringUtils.bore(parentData)
+    }
+    class Foo extends RawModule {
+      val a = IO(Input(UInt(1.W)))
+      val bar = Module(new Bar(a))
+    }
+    (circt.stage.ChiselStage
+      .emitCHIRRTL(new Foo)
+      .split('\n')
+      .map(_.takeWhile(_ != '@'))
+      .map(_.trim) should contain).inOrder(
+      "module Bar :",
+      "input q_bore : UInt<1>",
+      "q <= q_bore_1", // Do normal connection before secret ones
+      "q_bore_1 <= q_bore",
+      "module Foo :",
+      "input a : UInt<1>",
+      "bar.q_bore <= a"
+    )
+  }
+
+  it should "not affect DataMirror.getPorts, as they are secret (shhh)" in {
```

Comment:
this seems a little inconsistent that I am allowed to copy bored ports with toDefinition but I can't reflect on them even before .toDefinitioning? How am I supposed to interact with bored ports on a definition?

Reply:
Agreed, I'm fixing it to work the same way, that reflecting "fully closes the modules"